client/my-sites/site-settings/main.jsx

```diff
@@ -25,31 +24,21 @@ import SiteSettingsNavigation from './navigation';
 import SidebarNavigation from 'my-sites/sidebar-navigation';
 import JetpackDevModeNotice from './jetpack-dev-mode-notice';
 
-/**
- * Module vars
- */
-const debug = debugFactory( 'calypso:my-sites:site-settings' );
-
 export class SiteSettingsComponent extends Component {
-	constructor( props ) {
-		super( props );
-
-		// bound methods
-		this.updateSite = this.updateSite.bind( this );
-
-		this.state = {
-			site: this.props.sites.getSelectedSite()
-		};
-	}
 
-	componentWillMount() {
-		debug( 'Mounting SiteSettings React component.' );
-		this.props.sites.on( 'change', this.updateSite );
-	}
+	static propTypes = {
+		hasLoadedSitePurchasesFromServer: PropTypes.bool.isRequired,
+		purchasesError: PropTypes.object,
+		section: PropTypes.string,
+		sitePurchases: PropTypes.array.isRequired,
+		site: PropTypes.object,
```

Comment:
If you follow my suggestions for removing the `site` connected prop, don't forget to remove it from here, too.